File: unity-plugin/src/main/java/com/example/blehid/unity/events/PairingRequestedEvent.java
package com.example.blehid.unity.events;

import android.bluetooth.BluetoothDevice;

/**
 * Event fired when a device requests pairing.
 */
public class PairingRequestedEvent extends DeviceEvent {
    private final int pairingVariant;
    
    /**
     * Creates a new pairing requested event.
     * 
     * @param device The device requesting pairing
     * @param pairingVariant The pairing variant requested (PIN, passkey, etc.)
     */
    public PairingRequestedEvent(BluetoothDevice device, int pairingVariant) {
        super(EventType.PAIRING_REQUESTED, device);
        this.pairingVariant = pairingVariant;
    }
    
    /**
     * Creates a new pairing requested event.
     * 
     * @param deviceAddress The address of the device requesting pairing
     * @param pairingVariant The pairing variant requested (PIN, passkey, etc.)
     */
    public PairingRequestedEvent(String deviceAddress, int pairingVariant) {
        super(EventType.PAIRING_REQUESTED, deviceAddress);
        this.pairingVariant = pairingVariant;
    }
    
    /**
     * Gets the pairing variant requested.
     * 
     * @return Pairing variant
     */
    public int getPairingVariant() {
        return pairingVariant;
    }
    
    @Override
    public String toString() {
        return String.format("%s[pairingVariant=%d]", super.toString(), pairingVariant);
    }
}

File: unity-test/Assets/BleHidTest/Events/PairingRequestedEvent.cs
using System;

namespace BleHid.Events
{
    /// <summary>
    /// Event fired when a device requests pairing.
    /// </summary>
    public class PairingRequestedEvent : DeviceEvent
    {
        /// <summary>
        /// The pairing variant requested (PIN, passkey, etc.)
        /// </summary>
        public int PairingVariant { get; private set; }
        
        /// <summary>
        /// Creates a new pairing requested event.
        /// </summary>
        /// <param name="deviceAddress">MAC address of the device</param>
        /// <param name="pairingVariant">Pairing variant</param>
        /// <param name="eventId">ID of the event</param>
        /// <param name="javaTimestamp">Java timestamp (milliseconds since epoch)</param>
        public PairingRequestedEvent(string deviceAddress, int pairingVariant, Guid eventId, long javaTimestamp)
            : base(BleHidEventType.PairingRequested, deviceAddress, null, eventId, javaTimestamp)
        {
            PairingVariant = pairingVariant;
        }
        
        /// <summary>
        /// Creates a new pairing requested event.
        /// For events created on the Unity side.
        /// </summary>
        /// <param name="deviceAddress">MAC address of the device</param>
        /// <param name="pairingVariant">Pairing variant</param>
        public PairingRequestedEvent(string deviceAddress, int pairingVariant)
            : base(BleHidEventType.PairingRequested, deviceAddress)
        {
            PairingVariant = pairingVariant;
        }
        
        /// <summary>
        /// Gets a string representation of this event.
        /// </summary>
        public override string ToString()
        {
            return $"{base.ToString()}, pairingVariant={PairingVariant}";
        }
    }
}
